go/vt/vttest/local_cluster.go

```diff
@@ -294,9 +299,9 @@ func (db *LocalCluster) MySQLConnParams() mysql.ConnParams {
 
 func (db *LocalCluster) MySQLTCPConnParams() mysql.ConnParams {
 	connParams := db.mysql.Params(db.DbName())
-	_, port := db.mysql.Address()
+	host, port := db.mysql.Address()
 	connParams.UnixSocket = ""
-	connParams.Host = "127.0.0.1"
+	connParams.Host = host
```

Comment:
Prior to the new flags it seems we used `127.0.0.1` as the hostname. Shoudln't that be the default for the flags then instead of `0.0.0.0`

Reply:
I think mysql is binding to `0.0.0.0` by default (at least it is when run out of vttestserver).

Confirmed with this test:
```
$ /opt/vitess-19/bin/vttestserver --alsologtostderr --persistent_mode --mysql_bind_host=127.231.106.94 --grpc_bind_address=127.231.106.94 --vschema_ddl_authorized_users=% --enable_system_settings=false --mysql_server_version=8.0.34 --planner-version=gen4 --data_dir=/tmp/foobar/data --log_dir=/tmp/foobar/logs --port=15991 --tablet_hostname=127.231.106.94 --tablet_refresh_interval=10s --vtcombo-bind-host=127.231.106.94 --num_shards=1,1 --keyspaces=core_general_1_dev,core_general_1_test

// notice command with port 15991, but mysql listening on 15993 binds to 0.0.0.0

$ netstat -an | grep 1599 | grep LISTEN
tcp4       0      0  127.231.106.94.15991   *.*                    LISTEN
tcp46      0      0  *.15992                *.*                    LISTEN
tcp4       0      0  127.231.106.94.15994   *.*                    LISTEN
tcp46      0      0  *.15993                *.*                    LISTEN

$ lsof -i :15991-15994
COMMAND   PID             USER   FD   TYPE             DEVICE SIZE/OFF NODE NAME
mysqld  30509 gabrielparreiras   20u  IPv6 0x843d5de06a4e62a0      0t0  TCP *:15993 (LISTEN)
vtcombo 30513 gabrielparreiras   34u  IPv4 0x83b4c31e34aa900c      0t0  TCP localhost:15994 (LISTEN)
vtcombo 30513 gabrielparreiras   36u  IPv6 0x38b3535bf6c700cf      0t0  TCP *:15992 (LISTEN)
vtcombo 30513 gabrielparreiras   37u  IPv4 0x93e314ab7e1f052a      0t0  TCP localhost:15991 (LISTEN)
```